### PR TITLE
fix: Moved most subscriptions code to EE

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Subscriptions is part of Scale so the code should reflect that

## Changes

* Moved most Subscription code to EE

## How did you test this code?
* Automated tests

### CODE REVIEW

File: frontend/src/lib/components/Subscriptions/SubscriptionsModal.tsx [hunk 1]
```diff
@@ -16,24 +19,38 @@ export interface SubscriptionsModalProps extends SubscriptionBaseProps {
...
-                        onCancel={() => push(urlForSubscriptions(props))}
-                        onDelete={() => push(urlForSubscriptions(props))}
+                    <PayGatePage
+                        featureKey={AvailableFeature.SUBSCRIPTIONS}
+                        header={
+                            <>
+                                Introducing <span className="highlight">Subscriptions</span>!
+                            </>
+                        }
+                        caption="Get regular Insight or Dashboard reports directly to your inbox!"
```
Comment (pauldambra) [inline frontend/src/lib/components/Subscriptions/SubscriptionsModal.tsx:52]:
```suggestion
                        caption="Get Insight or Dashboard reports directly to your inbox!"
```

regular felt odd and most people will know what a subscription is?

File: posthog/api/unsubscribe.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import jwt
+from django.http import HttpRequest, JsonResponse
+
+from posthog.models.subscription import unsubscribe_using_token
+
+
+def unsubscribe(request: HttpRequest):
```
Comment (pauldambra) [inline posthog/api/unsubscribe.py:7]: should this be in`ee` too?

subscribe is `ee` but unsubscribe is `posthog` 🤔

**Review by pauldambra — APPROVED** [review]
two thoughts but not blocking

File: posthog/api/unsubscribe.py (hunk 2 above)
Comment (benjackwhite) [reply posthog/api/unsubscribe.py:7]: That's a good point. I guess my thinking was, if someone subscribes whilst on the Scale plan but downgrades later they should still be able to unsubscribe?

**Comment by benjackwhite** [comment]
@clarkus FYI this updates the Storybook code to be a bit more usable so you can easily see the different views
(https://61260360a8e859003a7c138b-xsilkvloai.chromatic.com/?path=/story/components-subscriptions--subscriptions) 

I'm going to merge this with the new paywall stuff so that we can safely do the self-hosted release but everything is still feature flagged.

File: posthog/api/unsubscribe.py (hunk 2 above)
Comment (pauldambra) [reply posthog/api/unsubscribe.py:7]: 🤯

**Pull request merged**